### PR TITLE
fix(lsm): Fixes for TigerBeetle LSM updates

### PR DESCRIPTION
**Pull request by koekiebox**
### Updates include:
* Initial changes for 'lsm' fixes.
* Updated error codes for create account and transfer.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export enum CreateAccountError {
   reserved_field,
 
   id_must_not_be_zero,
+  id_must_not_be_max_int,
```
Comment (jorangreef) [inline src/index.ts]:
```suggestion
  id_must_not_be_int_max,
```

We should change this upstream as well—would you please do that too @koekiebox ?

**Review by jorangreef — COMMENTED** [review]
Thanks @koekiebox !

File: src/index.ts (hunk 1 above)
Comment (koekiebox) [reply src/index.ts]: @jorangreef  Done. https://github.com/coilhq/tigerbeetle/pull/131

File: src/node.zig [hunk 2]
```diff
@@ -138,26 +139,19 @@ const Context = struct {
...
-        context.message_bus = try MessageBus.init(
-            allocator,
-            cluster,
-            context.addresses,
-            client_id,
-            context.io,
-        );
-        errdefer context.message_bus.deinit();
-
+        var message_pool = try MessagePool.init(allocator, .client);
```
Comment (kprotty) [inline src/node.zig]: The client holds an address to the MessagePool, but the message pool is invalidated once this function returns.
Try storing the MessagePool inside the Context, making sure to also deinit() it later.

**Review by sentientwaffle — APPROVED** [review]

**Review by sentientwaffle — CHANGES_REQUESTED** [review]
Sorry, approved the wrong PR!

**Review by sentientwaffle — APPROVED** [review]
LGTM.

**Pull request merged**